gulpfile.js

```diff
@@ -41,7 +32,19 @@ gulp.task('install-azure-account', () => {
             }))
             .pipe(gulp.dest(extensionPath));
     } else {
-        console.log('Azure Account extension already installed.');
+        // We don't use console.log because we need to signal completion of this async task to gulp
+        return gulp.src('package.json')
+            .pipe(print(function () { return 'Azure Account extension already installed.'; }));
     }
 });
 
+
+const testTask = () => {
+    const env = process.env;
+    env.DEBUGTELEMETRY = 1;
+    env.MOCHA_reporter = 'mocha-junit-reporter';
+    env.MOCHA_FILE = path.join(__dirname, 'test-results.xml');
+    return cp.spawn('node', ['./node_modules/vscode/bin/test'], { stdio: 'inherit', env });
```

Comment:
This seems to be new for gulp 4.0. You can return the result of a `cp.spawn` and it just handles it correctly.

https://www.joezimjs.com/javascript/complete-guide-upgrading-gulp-4/ (search for spawn)

Keep in mind gulp 4.0 is still in preview. This could always change, but I think I'm okay relying on it since it gets rid of security warnings.

Reply:
Yeah, if it's listening, that's great.